stustapay/core/service/customer.py

```diff
@@ -48,10 +49,55 @@ def csv_export(customers_bank_data: list[CustomerBankData], output_path: str) ->
                     customer.account_name,
                     customer.iban,
                     customer.balance,
-                    "EUR",
+                    currency_ident,
                     f"StuStaCulum, TagID: {customer.user_tag_uid}",
                 ]
             )
+    return len(customers_bank_data)
+
+
+def sepa_export(
+    customers_bank_data: list[CustomerBankData],
+    output_path: str,
+    cfg: Config,
+    currency_ident: str,
+    execution_date: Optional[datetime.date],
+) -> int:
+    if len(customers_bank_data) == 0:
+        return 0
+
+    execution_date = execution_date or datetime.date.today() + datetime.timedelta(days=2)
+    iban = IBAN(cfg.customer_portal.sepa_config.sender_iban)
+    config = {
+        "name": cfg.customer_portal.sepa_config.sender_name,
+        "IBAN": iban.compact,
+        "BIC": str(iban.bic),
+        "batch": True,
+        "currency": currency_ident,  # ISO 4217
+    }
+    sepa = SepaTransfer(config, clean=True)
+
+    print(customers_bank_data)
+
+    for customer_bank_data in customers_bank_data:
+        payment = {
+            "name": customer_bank_data.account_name,
+            "IBAN": IBAN(customer_bank_data.iban).compact,
+            "BIC": str(IBAN(customer_bank_data.iban).bic),
+            "amount": round(customer_bank_data.balance * 100),  # in cents
+            "execution_date": execution_date,
+            "description": cfg.customer_portal.sepa_config.description.format(
+                user_tag_uid=customer_bank_data.user_tag_uid
+            ),
+        }
+        print(payment)
+        sepa.add_payment(payment)
+
+    sepa_xml = sepa.export(validate=True, pretty_print=True)
```

Comment:
since we have thousands of exports, maybe we can create and export them one-by-one instead of having a giant list first, if sepa.export supports this.